### PR TITLE
add user defined properties

### PR DESCRIPTION
**Pull request by qiaojialin**
resolve issue #8 
Enable user to add properties in tsfile.
Add props in TSFileMetaData, A Map<String, String> structure. User can define this in the json schema file or use TsFile class to add properties. And can read this properties from TsFile class.

Remove RecordReader from TsFile, replaced by QueryEngine.

Modify some wrong parameters in docs.

### CODE REVIEW

File: src/main/java/cn/edu/thu/tsfile/file/metadata/TSFileMetaData.java [hunk 1]
```diff
@@ -145,6 +151,7 @@ public FileMetaData convertToThrift() {
           rowGroupMetaDataListInThrift);
       metaDataInThrift.setCreated_by(createdBy);
       metaDataInThrift.setJson_metadata(jsonMetaData);
```
Comment (kr11) [inline src/main/java/cn/edu/thu/tsfile/file/metadata/TSFileMetaData.java]: 稍有偏离，这些下划线+小写字母的函数命名风格与java风格略相异，不知道是不是自动生成的？

Comment (kr11) [inline src/main/java/cn/edu/thu/tsfile/file/metadata/TSFileMetaData.java]: 如果要统一的话，看起来改动很大

File: src/test/java/cn/edu/thu/tsfile/file/metadata/utils/Utils.java [hunk 2]
```diff
@@ -233,7 +234,20 @@ public static void isFileMetaDataEqual(TSFileMetaData fileMetaDataInTSF,
           fileMetaDataInThrift.getTimeseries_list());
       Utils.isListEqual(fileMetaDataInTSF.getJsonMetaData(),
           fileMetaDataInThrift.getJson_metadata(), "json metadata");
-
+      
+      if(Utils.isTwoObjectsNotNULL(fileMetaDataInTSF.getProperties(), fileMetaDataInThrift.getProperties(), "user specificed properties")){
```
Comment (kr11) [inline src/test/java/cn/edu/thu/tsfile/file/metadata/utils/Utils.java]: user specificed properties
是否是：specified？

Comment (kr11) [inline src/test/java/cn/edu/thu/tsfile/file/metadata/utils/Utils.java]: 以下三处出现的地方均同

Comment (MyXOF) [inline src/test/java/cn/edu/thu/tsfile/file/metadata/utils/Utils.java]: 已经修改拼写错误

**Pull request merged**